src/components/HourChart/HourChart.jsx

```diff
@@ -105,149 +185,96 @@ export default class HourChart extends PureComponent {
-    const innerMargin = {
-      top: 20,
-      right: innerMarginRight,
-      bottom: 35,
-      left: innerMarginLeft,
-    };
-
-    const innerWidth = width - innerMargin.left - innerMargin.right;
-    const innerHeight = height - innerMargin.top - innerMargin.bottom;
-
-    const xMin = 0;
-    const xMax = innerWidth;
-    const yMin = innerHeight;
-    const yMax = 0;
-
-    // set up the domains based on extent. Use the prop if provided, otherwise calculate
-    const xDomain = [0, 23];
-    let yDomain = yExtent || d3.extent(data, d => d[yKey]);
-
-    // force 0 as the min in the yDomain if specified
-    if (forceZeroMin) {
-      yDomain = [0, yDomain[1]];
-    }
-
-    // use the props xScale if provided, otherwise compute it
-    if (!xScale) {
-      xScale = d3.scaleLinear().domain(xDomain).range([xMin, xMax]);
-    }
-    const yScale = d3.scaleLinear().domain(yDomain).range([yMin, yMax]);
-
-    const binWidth = (xMax - yMax) / 24;
-
-    // function to generate paths for each series
-    const line = d3.line()
-      .curve(d3.curveLinear)
-      .x((d) => xScale(d.hour) + (binWidth / 2))
-      .y((d) => yScale(d[yKey]));
-
-    // for highlighted dates
-    const highlightColor = 'rgb(245, 46, 113)';
-
-    return {
-      binWidth,
-      color,
-      dataByHour,
-      dataByDate,
-      data,
-      height,
-      highlightColor,
-      innerHeight,
-      innerMargin,
-      innerWidth,
-      line,
-      overallData,
-      width,
-      xScale,
-      yScale,
-      yKey,
-    };
-  }
-
-  /**
    * Update the d3 chart - this is the main drawing function
    *
    * @return {void}
    */
   update() {
-    this.renderCircles();
-    this.renderOverallLine();
-    this.renderAxes();
-    this.renderHighlight();
-  }
-
-  /**
-   * Callback for when the mouse hovers over a circle
-   *
-   * @param {Object} d The hovered over data point
-   * @return {void}
-   */
-  onCircleMouseEnter(d) {
-    const { onHighlightPoint } = this.props;
-    if (onHighlightPoint) {
-      onHighlightPoint(d);
-    }
-  }
-
-  /**
-   * Callback for when the mouse leaves a circle
-   *
-   * @param {Object} d The previously hovered over data point
-   * @return {void}
-   */
-  onCircleMouseLeave() {
-    const { onHighlightPoint } = this.props;
-    if (onHighlightPoint) {
-      onHighlightPoint(null);
-    }
+    this.updateCircles();
+    this.updateOverallLine();
+    this.updateAxes();
+    this.updateHighlight();
   }
 
   /**
    * Render the x and y axis components
    */
-  renderAxes() {
-    const { xScale, yScale, innerHeight, binWidth } = this.visComponents;
+  updateAxes() {
+    const { xScale, yScale, innerHeight, innerWidth, innerMargin, binWidth, yKey, yFormatter } = this.props;
     const xAxis = d3.axisBottom(xScale).tickSizeOuter(0);
     const yAxis = d3.axisLeft(yScale).tickSizeOuter(0);
 
+    // use default formatter unless retransmit_avg since we want percentages rendered
+    if (yKey === 'retransmit_avg') {
```

Comment:
yeah I wasn't happy doing this in the line chart either... but the alternative I think is having a separate formatter for axes and for values otherwise you get things like 10.0 12.0 14.0 in the axis instead of 10 12 14